src/eos_gates_windows.js

```diff
@@ -1,28 +1,88 @@
 
 const Lang = imports.lang;
 
+const Flatpak = imports.gi.Flatpak;
 const GLib = imports.gi.GLib;
 const Gio = imports.gi.Gio;
-const EosMetrics = imports.gi.EosMetrics;
+const Gtk = imports.gi.Gtk;
 
 const EosGates = imports.eos_gates;
+EosGates.setupEnvironment();
 
 const WHITELISTED_APPS = [
     // This is an unlikely binary name to run into, so just add it
     // in here for testing purposes...
     { processName: "com.endlessm.test-whitelist.exe" },
 ];
 
+function generateEndlessInstallerEntry() {
+    let entry = {
+        regex: /endless-installer.*.exe/,
+        overrideHelpMessage: _("You are already running Endless OS!")
+    };
+
+    // If we detect the presence of eos-installer.desktop, inform
+        // the user on how to actually install Endless OS.
+    if (!!Gio.DesktopAppInfo.new("eos-installer.desktop")) {
+        entry.overrideHelpMessage = _("You are already running Endless OS from live media!");
+        entry.desktopInfo = { id: 'eos-installer.desktop' },
+        entry.replacementInfo = {
+            appName: _("Endless OS Installer"),
+            description: _("You can install Endless OS by running the Endless OS Installer.")
+        }
+    }
+
+    return entry;
+}
+
+const FLATPAK_APPS = [
+    {
+        regex: /[Ss]potify[Ss]etup.*.exe/,
+        appName: _("Spotify"),
+        flatpakInfo: { remote: 'eos-apps', id: 'com.spotify.Client' }
+    },
+    {
+        regex: /[Ff]irefox\s+[Ss]etup.*.exe/,
+        appName: _("Firefox"),
+        flatpakInfo: { remote: 'eos-apps', id: 'org.mozilla.Firefox' }
+    },
+    {
+        regex: /.*[Ss]kype.*.exe/,
+        appName: _("Skype"),
+        flatpakInfo: { remote: 'eos-apps', id: 'com.microsoft.Skype' }
+    },
+    {
+        regex: /iTunes.*.exe/,
+        appName: _("Apple iTunes"),
+        flatpakInfo: { remote: 'eos-apps', id: 'com.spotify.Client' },
+        replacementInfo: {
+            appName: _("Spotify"),
+            description: _("Spotify is a streaming music service")
```

Comment:
I had a look into this and it seems like there is quite a lot of boilerplate that we would have to write to get this to work (not to mention that it would have to be run on every launch: https://github.com/endlessm/gnome-software/blob/master/plugins/flatpak/gs-flatpak.c#L217). I think for now it makes more sense to just hardcode the descriptions/app names we want and if we want to move this information into appstream itself we should provide an easier way of querying the database.